sky/cli.py

```diff
@@ -3969,6 +4050,395 @@ def spot_dashboard(port: Optional[int]):
+
+    def _down_service(name: str):
+        success_progress = False
+        try:
+            sky.serve_down(name, purge)
+        except RuntimeError as e:
+            message = (
+                f'{colorama.Fore.RED}Tearing down service {name!r}...failed. '
+                'Please manually clean up the replicas and then use --purge '
+                f'to clean up the controller.{colorama.Style.RESET_ALL}'
+                f'\nReason: {common_utils.format_exception(e)}.')
+        else:
+            message = (
+                f'{colorama.Fore.GREEN}Tearing down service {name!r}...done.'
+                f'{colorama.Style.RESET_ALL}')
+            success_progress = True
+
+        progress.stop()
+        click.echo(message)
+        if success_progress:
+            progress.update(task, advance=1)
+        progress.start()
+
+    with progress:
+        subprocess_utils.run_in_parallel(_down_service, service_names)
+        progress.live.transient = False
+        # Make sure the progress bar not mess up the terminal.
+        progress.refresh()
+
+
+@serve.command('logs', cls=_DocumentedCodeCommand)
+@click.option(
+    '--follow/--no-follow',
+    is_flag=True,
+    default=True,
+    help=('Follow the logs of the job. [default: --follow] '
+          'If --no-follow is specified, print the log so far and exit.'))
+@click.option('--controller',
+              is_flag=True,
+              default=False,
+              required=False,
+              help='Show the controller logs of this service.')
+@click.option('--load-balancer',
+              is_flag=True,
+              default=False,
+              required=False,
+              help='Show the load balancer logs of this service.')
+@click.argument('service_name',
+                required=True,
+                type=str,
+                **_get_shell_complete_args(_complete_service_name))
+@click.argument('replica_id', required=False, type=int)
+@usage_lib.entrypoint
+def serve_logs(
+    service_name: str,
+    follow: bool,
+    controller: bool,
+    load_balancer: bool,
+    replica_id: Optional[int],
+):
+    """Tail the log of a service.
+
+    Example:
+
+    .. code-block:: bash
+
+        # Tail the controller logs of a service
+        sky serve logs --controller [SERVICE_ID]
+        \b
+        # Print the load balancer logs so far and exit
+        sky serve logs --load-balancer --no-follow [SERVICE_ID]
+        \b
+        # Tail the logs of replica 1
+        sky serve logs [SERVICE_ID] 1
+    """
+    have_replica_id = replica_id is not None
+    if (controller + load_balancer + have_replica_id) != 1:
+        raise click.UsageError(
+            'One and only one of --controller, --load-balancer, '
+            '[REPLICA_ID] can be specified.')
+    core.serve_tail_logs(service_name,
+                         controller=controller,
+                         load_balancer=load_balancer,
+                         replica_id=replica_id,
```

Comment:
As mentioned before, I would prefer to have an enum to specify which log to show.